Directory.Packages.props

```diff
@@ -7,7 +7,7 @@
   <ItemGroup>
     <PackageVersion Include="Google.Protobuf" Version="[3.19.4,4.0)" />
     <PackageVersion Include="Grpc" Version="[2.44.0,3.0)" />
-    <PackageVersion Include="Grpc.Net.Client" Version="[2.45.0,3.0)" />
+    <PackageVersion Include="Grpc.Net.Client" Version="[2.52.0,3.0)" />
```

Comment:
Are you sure 2.52.0 contains the fix? Based on the PR you reference, it looks like when it was merged it's commit is tagged with 2.53.0 https://github.com/grpc/grpc/commit/29d8beee0ac2555773b2a2dda5601c74a95d6c10

Reply:
Oh, my mistake the fix was the PR is against the core grpc repo not Grpc.Net.Client.

Reply:
`2.52.0` is also the lowest available version of the package with no vulnerabilities.

Reply:
https://www.nuget.org/packages/Grpc.Net.Client this is the one we'll be using.
I guess the general rule here is to reference the lowest version that is not deprecated?

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/17327289/2a4afcd8-7d3c-49fd-af05-8e82f0244ed8)